lib/exchange_api_web/controllers/order_controller.ex

```diff
@@ -1,10 +1,116 @@
 defmodule ExchangeApiWeb.OrderController do
   use ExchangeApiWeb, :controller
 
-  def index(conn, _params) do
-    json(conn, %{status: "** shrug das orders **"})
+  action_fallback ExchangeApiWeb.FallbackController
+
+  def index(conn, %{"ticker_id" => ticker}) do
+    with {:ok , tick} <- get_ticker(ticker) do
+      {status, data} = Exchange.open_orders(tick)
+      json(conn, %{status: status, data: data})
+    end
+  end
+
+
+  def create(conn,
+              %{"order_id" => order_id,
+                "trader_id" => trader_id,
+                "side" => side,
+                "size" => size,
+                "price" => price,
+                "type" => type,
+                "ticker_id" => ticker} = params) do
+
+    time = Map.get(params, "exp_time", nil)
+    exp = get_time(time)
+
+    {side_status, side} = get_side(side)
+    {tick_status, tick} = get_ticker(ticker)
+    {type_status, type} = get_type(type)
+    {status, _value} = is_valid({side_status, side}, {tick_status, tick}, {type_status, type})
```

Comment:
this validation could be done as you process the side, ticker, time and type, chained in a `with` statement, and if one returns an error tuple, you don't even bother processing the rest